server/graphql/general/src/queries/log.rs

```diff
@@ -0,0 +1,119 @@
+use async_graphql::*;
+use graphql_core::{
+    generic_filters::EqualFilterStringInput,
+    map_filter,
+    pagination::PaginationInput,
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use graphql_types::types::{LogConnector, LogNodeType};
+use repository::{EqualFilter, PaginationOption};
+use repository::{LogFilter, LogSort, LogSortField};
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    log::get_logs,
+};
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq)]
+#[graphql(remote = "repository::LogSortField")]
+#[graphql(rename_items = "camelCase")]
+pub enum LogSortFieldInput {
+    Id,
+    LogType,
+    UserId,
+    RecordId,
+}
+
+#[derive(InputObject)]
+pub struct LogSortInput {
+    /// Sort query result by `key`
+    key: LogSortFieldInput,
+    /// Sort query result is sorted descending or ascending (if not provided the default is
+    /// ascending)
+    desc: Option<bool>,
+}
+
+#[derive(InputObject, Clone)]
+pub struct EqualFilterLogTypeInput {
+    pub equal_to: Option<LogNodeType>,
+    pub equal_any: Option<Vec<LogNodeType>>,
+    pub not_equal_to: Option<LogNodeType>,
+}
+
+#[derive(InputObject, Clone)]
+pub struct LogFilterInput {
+    pub id: Option<EqualFilterStringInput>,
+    pub log_type: Option<EqualFilterLogTypeInput>,
+    pub user_id: Option<EqualFilterStringInput>,
+    pub record_id: Option<EqualFilterStringInput>,
+}
+
+#[derive(Union)]
+pub enum LogResponse {
+    Response(LogConnector),
+}
+
+pub fn logs(
+    ctx: &Context<'_>,
+    store_id: String,
+    page: Option<PaginationInput>,
+    filter: Option<LogFilterInput>,
+    sort: Option<Vec<LogSortInput>>,
+) -> Result<LogResponse> {
+    validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::QueryItems,
```

Comment:
Is this the correct resource? better to introduce a QueryLog resource and then assign the required permissions to it (`service/src/auth.rs`). There is a mSupply ViewLog permission in `service/src/apis/permissions.rs`. The mSupply to omSupply permission mapping is done in: `service/src/login.rs`.